projects/common/src/time/time.ts

```diff
@@ -32,11 +32,27 @@ export class Time {
     return this._date;
   }
 
+  public static parse(label: string, isUTC: boolean = false): Time {
+    const time: string[] = label.split(' ');
+    const meridian: string = time.pop()!;
+    const [hours, minutes, seconds, milliseconds] = time[0].split(':').map(value => +value);
+
+    return new Time(Time.get24HrClockHours(hours, meridian), minutes, seconds, milliseconds, isUTC);
+  }
+
   public toISOString(): string {
     return this.date.toISOString().substring(11);
   }
 
   public equals(other?: Time): boolean {
     return this.toISOString() === other?.toISOString();
   }
+
+  private static get24HrClockHours(hours: number, meridian: string): number {
+    if (meridian === 'AM') {
+      return hours === 12 ? 0 : hours;
+    }
+
+    return hours === 12 ? 12 : hours + 12;
+  }
```

Comment:
can be shortened to

```
return meridian === 'AM' ? hour %12 : hour + 12
```

Reply:
oh yeah, mods

Reply:
updated

Reply:
> can be shortened to
> 
> ```
> return meridian === 'AM' ? hour %12 : hour + 12
> ```

This will fail at 12:30 PM, updated to 
`return meridian === 'AM' ? hours % 12 : hours === 12 ? 12 : hours + 12;`